js/src/utils/jestlike/index.ts

```diff
@@ -0,0 +1,612 @@
+                  syncExamplePromises.set(
+                    exampleId,
+                    await syncExample({
+                      client,
+                      exampleId,
+                      datasetId: dataset.id,
+                      inputs,
+                      outputs: expected,
+                      metadata: {},
+                      createdAt,
+                    })
+                  );
+                }
+
+                const traceableOptions = {
+                  reference_example_id: exampleId,
+                  project_name: project!.name,
+                  metadata: {
+                    ...config?.metadata,
+                  },
+                  client,
+                  tracingEnabled: true,
+                  name,
+                };
+
+                // Pass inputs into traceable so tracing works correctly but
+                // provide both to the user-defined test function
+                const tracedFunction = traceable(
+                  async () => {
+                    return testWrapperAsyncLocalStorageInstance.run(
+                      {
+                        ...context,
+                        currentExample: {
+                          inputs,
+                          outputs: expected,
+                          id: exampleId,
+                        },
+                        setLoggedOutput,
+                        onFeedbackLogged,
+                      },
+                      runTestFn
+                    );
+                  },
+                  {
+                    ...traceableOptions,
+                    ...config,
+                  }
+                );
+                try {
+                  await tracedFunction(testInput);
+                } catch (e: any) {
+                  // Extract raw Jest error from LangSmith formatted one and throw
+                  if (e.rawJestError !== undefined) {
+                    throw e.rawJestError;
+                  }
+                  throw e;
+                }
+              } else {
+                try {
+                  await testWrapperAsyncLocalStorageInstance.run(
+                    {
+                      ...context,
+                      currentExample: {
+                        inputs: testInput,
+                        outputs: testOutput,
+                      },
+                      setLoggedOutput,
+                      onFeedbackLogged,
+                    },
+                    runTestFn
+                  );
+                } catch (e: any) {
+                  // Extract raw Jest error from LangSmith formatted one and throw
+                  if (e.rawJestError !== undefined) {
+                    throw e.rawJestError;
+                  }
+                  throw e;
+                }
+              }
+            } finally {
+              await fs.mkdir(path.dirname(resultsPath), { recursive: true });
```

Comment:
Legitimately could not figure out another way to pass test results to Jest except via the filesystem or something heavyweight like this:

https://github.com/wix-incubator/jest-metadata